packages/contracts/src/example/AnonAadhaarVote.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.19;
+
+import "../../interfaces/IAnonAadhaar.sol";
+import "../../interfaces/IAnonAadhaarVote.sol";
+
+contract AnonAadhaarVote is IAnonAadhaarVote {
+    string public votingQuestion;
+    address public anonAadhaarVerifierAddr;
+
+    // List of proposals
+    Proposal[] public proposals;
+
+    // Mapping to track if a nullifier has already voted
+    // userNullifier can be accessed in _pubInputs => _pubInputs[1]
+    mapping(uint256 => bool) public hasVoted;
+
+    // Constructor to initialize proposals
+    constructor(string memory _votingQuestion, string[] memory proposalDescriptions, address _verifierAddr) {
+        anonAadhaarVerifierAddr = _verifierAddr;
+        votingQuestion = _votingQuestion;
+        for (uint256 i = 0; i < proposalDescriptions.length; i++) {
+            proposals.push(Proposal(proposalDescriptions[i], 0));
+        }
+    }
+
+    /// @dev Convert an address to uint256, used to check against signal.
+    /// @param _addr: msg.sender address.
+    /// @return Address msg.sender's address in uint256
+    function addressToUint256(address _addr) private pure returns (uint256) {
+        return uint256(uint160(_addr));
+    }
+
+    /// @dev Register a vote in the contract.
+    /// @param proposalIndex: Index of the proposal you want to vote for.
+    /// @param identityNullifier: Hash of last the 4 digits + DOB, name, gender adn pin code.
+    /// @param userNullifier: Hash of the last 4 digits + photo.
+    /// @param timestamp: Timestamp of when the QR code was signed.
+    /// @param signal: signal used while generating the proof, should be equal to msg.sender.
+    /// @param groth16Proof: SNARK Groth16 proof.
+    function voteForProposal(uint256 proposalIndex, uint identityNullifier, uint userNullifier, uint timestamp, uint signal, uint[8] memory groth16Proof ) public {
+        require(proposalIndex < proposals.length, "[AnonAadhaarVote]: Invalid proposal index");
+        require(addressToUint256(msg.sender) == signal, "[AnonAadhaarVote]: wrong user signal sent.");
+        require(IAnonAadhaar(anonAadhaarVerifierAddr).verifyAnonAadhaarProof(identityNullifier, userNullifier, timestamp, signal, groth16Proof) == true, "[AnonAadhaarVote]: proof sent is not valid.");
```

Comment:
Should we check timestamp here?

Reply:
That's a good point!
I wonder if it's something that should be implemented here, or just should be specified in the documentation, as it will be hard for us to have a proper test environment to test it.

Reply:
We can do it later. 
I remember in hardhat we can set timestamp so maybe this help we test smart contract easier.

Reply:
Good points. If its difficult for tests we can simply add a comment and commented out code to handle timestamp. Since this is an example others referring to it can implement properly.

Reply:
I added it without making the test fail as suggested by @vuvoth. Just changed the starting block of the EVM fork running while testing.